src/runtime/virtcontainers/clh.go

```diff
@@ -287,12 +287,14 @@ func (clh *cloudHypervisor) createSandbox(ctx context.Context, id string, networ
 
 	// set the serial console to the cloud hypervisor
 	if clh.config.Debug {
-		clh.vmconfig.Serial = chclient.NewConsoleConfig(cctTTY)
+		clh.vmconfig.Console = chclient.NewConsoleConfig(cctTTY)
+		clh.vmconfig.Console.SetMode("Tty")
+		clh.vmconfig.Console.SetIommu(true)
 	} else {
-		clh.vmconfig.Serial = chclient.NewConsoleConfig(cctNULL)
+		clh.vmconfig.Console = chclient.NewConsoleConfig(cctNULL)
 	}
 
-	clh.vmconfig.Console = chclient.NewConsoleConfig(cctOFF)
+	clh.vmconfig.Serial = chclient.NewConsoleConfig(cctOFF)
```

Comment:
I'd set the serial as `Null` instead of `Off`. Using `Off` means the device won't be emulated at all, which might block the guest kernel boot as it might still try to write a few bytes to the serial port.
Using `Null` emulates the device but anything that gets written goes to `/dev/null`.

Reply:
Good call - fixed.